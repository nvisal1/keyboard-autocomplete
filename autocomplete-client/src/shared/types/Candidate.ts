export type Candidate = {
    word: string;
    confidence: number;
}