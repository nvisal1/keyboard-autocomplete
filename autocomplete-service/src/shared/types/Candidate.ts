export class Candidate {

    private _word: string;
    private _confidence: number;

    constructor(word: string, confidence: number) {
        this._word = word;
        this._confidence = confidence;
    }

    getWord(): string {
        return this._word;
    }

    getConfidence(): number {
        return this._confidence;
    }
}