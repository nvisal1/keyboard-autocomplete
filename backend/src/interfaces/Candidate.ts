export interface Candidate {
    getWord() : Promise<string>; //returns the autocomplete candidate
    getConfidence() : Promise<number> // returns the confidence* for the candidate
}