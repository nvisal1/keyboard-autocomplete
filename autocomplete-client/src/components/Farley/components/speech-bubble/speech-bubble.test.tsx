import React from 'react';
import renderer from 'react-test-renderer';

import SpeechBubble, { SpeechBubbleProps } from './speech-bubble';
import { Candidate } from '../../../../shared/types/Candidate';

describe('When the SpeechBubble component is created', () => {
    describe('and the errorMessage prop is set to null', () => {
        it('should render correctly', () => {
            const requiredProps = getRequiredProps();
            const currentSnapshot = renderer.create(<SpeechBubble { ...requiredProps }></SpeechBubble>).toJSON();

            expect(currentSnapshot).toMatchSnapshot();
        });
    });
    describe('and the errorMessage prop is set to a truthy string', () => {
        it('should render correctly', () => {
            const requiredProps = getRequiredProps();
            const propsWithErrorMessage = { ...requiredProps, errorMessage: 'test_error_messagge' };
            const currentSnapshot = renderer.create(<SpeechBubble { ...propsWithErrorMessage }></SpeechBubble>).toJSON();

            expect(currentSnapshot).toMatchSnapshot();
        });
    });
});

function getRequiredProps(): SpeechBubbleProps {
   const handleClick = (_: string) => {} 

   const candidates: Candidate[] = [{ word: 'test_word', confidence: 1 }]; 

   return { handleClick, candidates };
}