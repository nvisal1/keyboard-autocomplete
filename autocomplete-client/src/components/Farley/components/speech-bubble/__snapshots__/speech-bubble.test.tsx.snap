// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the SpeechBubble component is created and the errorMessage prop is set to a truthy string should render correctly 1`] = `
<div
  className="speech-bubble"
>
  <div
    className="speech-bubble__text-container"
  >
    <p
      className="speech-bubble__text-container__error-message"
    >
      test_error_messagge
    </p>
    <div
      className="speech-bubble__pointer-container"
    />
  </div>
</div>
`;

exports[`When the SpeechBubble component is created and the errorMessage prop is set to null should render correctly 1`] = `
<div
  className="speech-bubble"
>
  <div
    className="speech-bubble__text-container"
  >
    <p
      className="speech-bubble__text-container__word"
      onClick={[Function]}
    >
      test_word(1)
    </p>
    <div
      className="speech-bubble__pointer-container"
    />
  </div>
</div>
`;
